File: javascript/CrappyScriptGenerator.js
var XLSX = require('xlsx');
var applicationInformation = [
    {
        columnIndex : 1,
        applicationId : 2237
    },
    {
        columnIndex : 2,
        applicationId : 4352
    },
    {
        columnIndex : 3,
        applicationId : 3657
    },
    {
        columnIndex : 4,
        applicationId : 5565
    }
];

function getInsertStatement(userId, applicationId) {
    return "insert into ApplicationPermission(user_id, application_id) values('"
        + userId
        + "', "
        + applicationId
        + ");";
}

function generate() {
    let lines = [];
    let workbook = XLSX.readFile('Users.xlsx');
    let worksheet = workbook.Sheets[workbook.SheetNames[0]];
    let sheetData = XLSX.utils.sheet_to_json(worksheet, { header: 1 }); // generate an array of arrays
    sheetData.forEach(function (row) {
        let userId = row[0];
        if (userId !== undefined) {
            if (userId.substring(1, 2) === '.') {
                // valid user id
                applicationInformation.forEach(function (appInfo) {
                    let value = row[appInfo.columnIndex];
                    if (value === 'X') {
                        lines.push(getInsertStatement(userId, appInfo.applicationId));
                    }
                });
            }
        }
    });
    return lines;
}

var lines = generate();
lines.forEach(line => console.log(line));

File: csharp/CrappyScriptGenerator.cs
using System;
using System.Collections.Generic;
using System.IO;
using OfficeOpenXml;

namespace delete_your_for_loops
{
    class CrappyScriptGenerator
    {
        static void Main(string[] args)
        {
            int[,] applicationInformation =
            {
                {2, 2257}, {3, 4352}, {4, 3657}, {5, 5565}
            };
            List<string> lines = new List<string>();

            FileInfo file = new FileInfo("Users.xlsx");
            ExcelPackage package = new ExcelPackage(file);
            ExcelWorksheet sheet = package.Workbook.Worksheets[1];
            int rows = sheet.Dimension.Rows;

            for (int row = 1; row <= rows; row++)
            {
                object value = sheet.Cells[row, 1].Value;
                if (value != null)
                {
                    string userId = value.ToString();
                    if (userId.Substring(1, 1) == ".")
                    {
                        for (int i = 0; i < applicationInformation.GetLength(0); i++)
                        {
                            value = sheet.Cells[row, applicationInformation[i, 0]].Value;
                            if (value != null)
                            {
                                String cellValue = value.ToString();
                                if (cellValue == "X")
                                {
                                    lines.Add(getInsertStatement(userId, applicationInformation[i, 1]));
                                }
                            }
                        }
                    }
                }
            }

            Console.WriteLine(lines.Count + " lines generated");
            foreach(string line in lines)
            {
                Console.WriteLine(line);
            }
        }

        private static string getInsertStatement(string userId, int applicationId)
        {
            return "insert into ApplicationPermission(user_id, application_id) values('"
            + userId
            + "', "
            + applicationId
            + ");";
        }
    }
}
